api/users/serializers.py

```diff
@@ -11,6 +11,29 @@
 )
 from api.base.utils import absolute_reverse
 
+class PublicFilesSerializer(JSONAPISerializer):
```

Comment:
It wouldn't necessarily be dependent on a PublicFiles endpoint -- a subclass of `FileSerializer` that could optionally force guid creation (assuming the user has proper permissions) wouldn't need to be limited to association with only one view. Also, subclassing would only provide benefits over adding that in `FileSerializer` if there are certain file views that we never want to allow the forced creation of guids for. I'm not sure that there are any, but it seems probable.

And it is likely that increasing guid length for files will be a blocker for this feature, but thankfully that shouldn't be hard to do. There's a `__guid_min_length__` that can be set on `GuidStoredObject`s, and it doesn't look like it would break backwards compatibility if it was suddenly bumped up.